packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart

```diff
@@ -731,3 +734,89 @@ List<MapEntry<String, WidgetBuilder>> vmObjectGeneralDataRows(
       ),
   ];
 }
+
+/// Creates a simple [CodeView] which displays the code relevant to [object] in
+/// [script]. If [object] is synthetic and doesn't have actual token positions,
+/// [object]'s owner's code will be displayed instead.
+class ObjectInspectorCodeView extends StatefulWidget {
+  ObjectInspectorCodeView({
+    required this.debuggerController,
+    required this.script,
+    required this.object,
+  }) : super(key: ValueKey(object));
+
+  final DebuggerController debuggerController;
+  final ScriptRef script;
+  final ObjRef object;
+
+  @override
+  State<ObjectInspectorCodeView> createState() =>
+      _ObjectInspectorCodeViewState();
+}
+
+class _ObjectInspectorCodeViewState extends State<ObjectInspectorCodeView> {
+  @override
+  void didChangeDependencies() async {
+    super.didChangeDependencies();
+    if (widget.script != widget.debuggerController.currentScriptRef.value) {
+      widget.debuggerController.resetScriptLocation(
+        ScriptLocation(widget.script),
+      );
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<ParsedScript?>(
+      valueListenable: widget.debuggerController.currentParsedScript,
+      builder: (context, currentParsedScript, _) {
+        LineRange? lineRange;
+        if (currentParsedScript != null) {
+          final obj = widget.object;
+          SourceLocation? location;
+          if (obj is ClassRef || obj is FuncRef || obj is FieldRef) {
+            final dynObj = obj as dynamic;
```

Comment:
avoid `as dynamic`

Reply:
Yeah, I typically do, but the alternative isn't great either:

```dart
if (obj is ClassRef) {
  location = object.location;
} else if (obj is FuncRef) {
  location = object.location;
} ...
```

Reply:
even though that has duplicate code I think it is better and may even trigger lints to leverage patterns in the future.